src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -23,6 +25,28 @@ open Microsoft.Quantum.QsCompiler.Transformations.QsCodeOutput
 
 // utils for verifying types in expressions
 
+/// The context used for symbol resolution and type checking.
+type ResolutionContext<'a> =
+    { /// The symbol tracker for the parent callable.
+      Symbols : SymbolTracker<'a>
+      /// True if the parent callable for the current scope is an operation.
+      IsInOperation : bool
+      /// The return type of the parent callable for the current scope.
+      ReturnType : ResolvedType
+      /// The runtime capabilities for the compilation unit.
+      Capabilities : RuntimeCapabilities }
+
+/// <summary>Creates a resolution context for the specialization.</summary>
```

Comment:
As much as I avoid any kind of redundancy in code, I would not do that for doc comments. The doc comment for the symbol tracker would be good to have here as well, since it directly applies to this method: 

 ```suggestion
/// <summary>
/// Creates a resolution context for the specialization.
/// The symbol tracker initialized as part of the context does *not* make a copy of the given namespace manager, 
/// but instead will throw an InvalidOperationException upon accessing its content if that content has been modified 
/// (i.e. if the version number of the namespace manager has changed). 
/// </summary>
```